src/Lumper.UI/ViewModels/MainWindowViewModel.cs

```diff
@@ -1,92 +1,194 @@
-﻿using System;
-using Avalonia;
-using Avalonia.Controls.ApplicationLifetimes;
-using Lumper.UI.ViewModels.Bsp;
-using Lumper.UI.ViewModels.Tasks;
-using Lumper.UI.ViewModels.VtfBrowser;
-using ReactiveUI;
-
 namespace Lumper.UI.ViewModels;
 
-/// <summary>
-///     ViewModel for MainWindow
-/// </summary>
-public partial class MainWindowViewModel : ViewModelBase
+using System;
+using System.Collections.Generic;
+using System.Reactive.Linq;
+using System.Threading.Tasks;
+using Avalonia.Controls;
+using Avalonia.Platform.Storage;
+using LogViewer;
+using MsBox.Avalonia;
+using MsBox.Avalonia.Base;
+using MsBox.Avalonia.Dto;
+using MsBox.Avalonia.Enums;
+using MsBox.Avalonia.Models;
+using NLog;
+using ReactiveUI;
+using Services;
+
+public class MainWindowViewModel : ViewModel
 {
-    private ViewModelBase? content;
-    private BspNodeBase? _selectedNode;
-    private BspViewModel? _bspModel;
-    private TasksViewModel? _tasksModel;
-    private VtfBrowserViewModel? _vtfBrowserModel;
+    public static BspService BspService => BspService.Instance;
+    public static PageService PageService => PageService.Instance;
+
+    public LogViewerViewModel LogViewer { get; set; } = new();
+
+    private static readonly Logger Logger = LogManager.GetCurrentClassLogger();
 
     public MainWindowViewModel()
     {
-        if (Application.Current?.ApplicationLifetime is not
-            IClassicDesktopStyleApplicationLifetime
-            desktop)
-            throw new InvalidCastException(
-                nameof(Application.Current.ApplicationLifetime));
+        if (Program.Desktop.Args is { Length: 1 })
+        {
+            Observable.Start(
+                () => BspService.Instance.Load(Program.Desktop.Args[0]),
+                RxApp.MainThreadScheduler);
+        }
+    }
+    /// <summary>
+    /// Event called when 'check for updates' is pressed in the File menu that triggers checking for and installing updates
+    /// </summary>
+    public async ValueTask UpdateCommand()
+    {
+        Lumper.UI.Updater.Updater.Version? updateAvailable = await Lumper.UI.Updater.Updater.CheckForUpdate();
+        if (updateAvailable != null)
+        {
+            string updateNumber = updateAvailable.ToString();
+            ButtonResult result = await MessageBoxManager
+                .GetMessageBoxStandard(
+                    "Update Available",
+                    $"An update is available ({updateNumber}), do you want to download and restart?", ButtonEnum.OkCancel)
+                .ShowWindowDialogAsync(Program.Desktop.MainWindow);
```

Comment:
Actually sorry, I suggested returning null here but I think a bool would be *way* easier to understand. So make method signature would be 
`bool CheckForUpdate(out Version version)`

then in the method have something like
```cs
version = latest;
return (current != latest);
```

then here would be
```cs
        var updateAvailable = await Lumper.UI.Updater.Updater.CheckForUpdate(out Version version);
        if (updateAvailable)
        {
             ButtonResult result = await MessageBoxManager
                .GetMessageBoxStandard(
                    "Update Available",
                    $"An update is available ({version), do you want to download and restart?", ButtonEnum.OkCancel)
                .ShowWindowDialogAsync(Program.Desktop.MainWindow);
```

Also import is quite weird, just do `import Lumper.UI.Updater` then you can do just e.g. `Updater.CheckForUpdate()`